internal/gengapic/gengapic.go

```diff
@@ -52,82 +51,164 @@ const (
 
 var headerParamRegexp = regexp.MustCompile(`{([_.a-z]+)`)
 
-func Gen(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, error) {
-	var pkgPath, pkgName, outDir string
-	var g generator
+type Transport int
+
+const (
+	grpc Transport = iota
+	rest
+)
+
+type options struct {
+	pkgPath           string
+	pkgName           string
+	outDir            string
+	relLvl            string
+	modulePrefix      string
+	grpcConfPath      string
+	serviceConfigPath string
+	sampleOnly        bool
+	transports        []Transport
```

Comment:
I believe Java assumes that either gRPC or REST will be used at a particular time. Does this array imply that callers can pass in both transport types?

Reply:
Yes, that is the implication. [Python](https://github.com/googleapis/gapic-generator-python/pull/688) allows multiple transports to be generated for the surface with runtime choice via dependency injection. This is supported by the spec: see [this](https://docs.google.com/document/d/1u0WGcF6woB9HnyUK5J-GaBCRUc9G4znwbvcB4QdYq8s/edit#heading=h.lebdm4farlvy) heading.